packages/cloudflare/src/cli/build/index.ts

```diff
@@ -178,3 +180,79 @@ function ensureCloudflareConfig(config: OpenNextConfig) {
     );
   }
 }
+
+/**
+ * Creates a `wrangler.toml` file for the user if it doesn't exist, but only after asking for the user's confirmation.
+ *
+ * If the user refuses an error is thrown (since the file is mandatory).
+ *
+ * @param projectOpts The options for the project
+ */
+async function createWranglerTomlIfNotExistent(projectOpts: ProjectOptions): Promise<void> {
+  const wranglerTomlPath = join(projectOpts.sourceDir, "wrangler.toml");
+
+  if (!existsSync(wranglerTomlPath)) {
+    const answer = await askConfirmation("Missing required `wrangler.toml` file, do you want to create one?");
+
+    if (!answer) {
+      throw new Error("The `wrangler.toml` file is required, aborting!");
+    }
+
+    const wranglerTomlTemplate = readFileSync(
+      join(getPackageTemplatesDirPath(), "defaults", "wrangler.toml"),
+      "utf8"
+    );
+    let wranglerTomlContent = wranglerTomlTemplate;
+
+    const appName = getAppNameFromPackageJson(projectOpts.sourceDir) ?? "app-name";
+    if (appName) {
+      wranglerTomlContent = wranglerTomlContent.replace(
+        '"app-name"',
+        JSON.stringify(appName.replaceAll("_", "-"))
+      );
+    }
+
+    const compatDate = await getLatestCompatDate();
+    if (compatDate) {
+      wranglerTomlContent = wranglerTomlContent.replace(
+        /compatibility_date = "\d{4}-\d{2}-\d{2}"/,
+        `compatibility_date = ${JSON.stringify(compatDate)}`
+      );
+    }
+
+    writeFileSync(wranglerTomlPath, wranglerTomlContent);
+  }
+}
+
+function getAppNameFromPackageJson(sourceDir: string): string | undefined {
+  try {
+    const packageJsonStr = readFileSync(join(sourceDir, "package.json"), "utf8");
+    const packageJson: Record<string, string> = JSON.parse(packageJsonStr);
+    if (typeof packageJson.name === "string") return packageJson.name;
+  } catch {
+    /* empty */
+  }
+}
+
+export async function getLatestCompatDate(): Promise<string | undefined> {
+  try {
+    const resp = await fetch(`https://registry.npmjs.org/workerd`);
+    const latestWorkerdVersion = (
+      (await resp.json()) as {
+        "dist-tags": { latest: string };
+      }
+    )["dist-tags"].latest;
+
+    // The format of the workerd version is `major.yyyymmdd.patch`.
+    const match = latestWorkerdVersion.match(/\d+\.(\d{4})(\d{2})(\d{2})\.\d+/);
+
+    if (match) {
+      const [, year, month, date] = match ?? [];
```

Comment:
Nit: ?? Should not be needed?

Reply:
good callout, no, I think that that's totally unnecessary 👍 

I just copied the code from https://github.com/cloudflare/workers-sdk/blob/2e78812ade7cd7361b023c90afe06221a52b79eb/packages/create-cloudflare/src/helpers/compatDate.ts#L30 without looking into it much 😅

Reply:
done 👍  (a28f9976a7c6344ad4788bdb7ca91fa5e9eb5ed5)

Reply:
I've copied this minor cleanup to the original `workers-sdk` codebase: https://github.com/cloudflare/workers-sdk/pull/7658